core/src/IPlatformProxy.cs

```diff
@@ -35,6 +35,24 @@ internal interface IPlatformProxy
 
         Task<bool> IsUserLocalAsync(RequestContext requestContext);
 
+        /// <summary>
+        /// Returns the name of the calling assembly
+        /// </summary>
+        /// <returns></returns>
+        string GetApplicationName();
```

Comment:
I would name them as GetCallingAseemblyName and GetCallingAssemblyVersion to differentiate better with the GetLibraryName etc.

Reply:
Done